lib/pbench/test/unit/server/database/test_index_map_db.py

```diff
@@ -90,14 +90,18 @@ def test_merge(self, db_session, attach_dataset, m1, m2):
             IndexStream("prefix.run-data.2023-06", "id9"),
         ]
 
-    @pytest.mark.parametrize("to", (False, True))
-    def test_merge_none(self, db_session, attach_dataset, to):
+    @pytest.mark.parametrize(
+        "orig,merge", ((False, True), (True, False), (False, False))
+    )
+    def test_merge_identity(self, db_session, attach_dataset, orig, merge):
         """Test index map merge with an empty map
 
-        We test two cases: one merging a map into an empty map, and then
-        merging an empty map into an existing map.
+        We test four cases: one merging a map into an empty map, then merging
+        an empty map into an existing map, merging a map into itself, and
+        finally merging two empty maps.
 
-        In either case the result should be the single map.
+        In "empty into empty" the result should be empty; in all other cases
+        the result should be the single map.
         """
```

Comment:
Contrary to what line 99 says, I think we're testing only three cases -- I don't think we're testing the case of merging a map into itself.

Reply:
Right: because I wrote the comment and then the test cases before I realized that self-merge duplicates all the document IDs and, if I cared enough to test that "impossible" case I'd need an entirely different validation which wouldn't really fit under the reasonable parametrization... but then I forgot to delete the ghost case from the comment.